tests/net/mgmt/src/mgmt.c

```diff
@@ -423,4 +423,32 @@ ZTEST(mgmt_fn_test_suite, test_mgmt)
 		      "test_synchronous_event_listener failed");
 }
 
+K_SEM_DEFINE(wait_for_event_processing, 0, 1);
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	ARG_UNUSED(cb);
+	ARG_UNUSED(iface);
+	ARG_UNUSED(mgmt_event);
+
+	k_sem_give(&wait_for_event_processing);
+}
+
+ZTEST(mgmt_fn_test_suite, test_mgmt_duplicate_handler)
+{
+	struct net_mgmt_event_callback cb;
+	int ret;
+
+	net_mgmt_init_event_callback(&cb, net_mgmt_event_handler, 0);
+	net_mgmt_add_event_callback(&cb);
+	net_mgmt_init_event_callback(&cb, net_mgmt_event_handler, 0);
+	net_mgmt_add_event_callback(&cb);
+
+	ret = k_sem_take(&wait_for_event_processing, K_SECONDS(1));
+	zassert_equal(ret, -EAGAIN, "Event is not processed");
+
+	/* If we reach here, the test is successful */
+	zassert_true(true, NULL);
```

Comment:
This is not needed, we do not need to check that test is ok, but only when it fails.